timed/tracking/signals.py

```diff
@@ -15,20 +15,28 @@ def update_rejected_on_reports(sender, instance, **kwargs):
             instance.rejected = False
 
 
-@receiver(post_save, sender=Report)
+@receiver(pre_save, sender=Report)
 def update_most_recent_remaining_effort(sender, instance, **kwargs):
     """Update remaining effort on task, if remaining effort tracking is active."""
-    if instance.task.project.remaining_effort_tracking:
-        task = instance.task
-        last_report = task.reports.order_by("date").last()
-        if instance == last_report:
-            task.most_recent_remaining_effort = instance.remaining_effort
-            task.save()
-        project = task.project
-        total_remaining_effort = (
-            project.tasks.all()
-            .aggregate(sum_remaining=Sum("most_recent_remaining_effort"))
-            .get("sum_remaining")
-        )
-        project.total_remaining_effort = total_remaining_effort
-        project.save()
+    # ignore signal when loading a fixture
+    if kwargs.get("raw", False):  # pragma: no cover
+        return
+
+    if instance.pk:
+        if instance.task.project.remaining_effort_tracking:
```

Comment:
It's more agreeable when reading code if unmet conditions are ruled out immediately aka free up memory where possible and stay a close to the left border as possible.

```suggestion
    if not instance.pk:
        return
    if instance.task.project.remaining_effort_tracking is not True:
        return
```

Reply:
makes sense, thanks :)